bridge/client/app/_components/ktb-trigger-sequence/ktb-trigger-sequence.component.ts

```diff
@@ -0,0 +1,351 @@
+      stage: this.selectedStage || '',
+      service: this.selectedService || '',
+      evaluation: {},
+    };
+    if (this.evaluationFormData.labels && this.evaluationFormData.labels.trim() !== '') {
+      data.evaluation.labels = this.parseLabels(this.evaluationFormData.labels);
+    }
+
+    if (
+      this.evaluationFormData.evaluationType === TRIGGER_EVALUATION_TIME.TIMEFRAME &&
+      this.evaluationFormData.timeframe
+    ) {
+      data.evaluation.timeframe = this.parseTimeframe(this.evaluationFormData.timeframe);
+      data.evaluation.start =
+        this.evaluationFormData.timeframeStart && this.evaluationFormData.timeframeStart !== ''
+          ? moment(this.evaluationFormData.timeframeStart).toISOString()
+          : moment().toISOString();
+    }
+
+    if (this.evaluationFormData.evaluationType === TRIGGER_EVALUATION_TIME.START_END) {
+      data.evaluation.start =
+        this.evaluationFormData.startDatetime && this.evaluationFormData.startDatetime !== ''
+          ? moment(this.evaluationFormData.startDatetime).toISOString()
+          : moment().toISOString();
+      data.evaluation.end = moment(this.evaluationFormData.endDatetime).toISOString();
+    }
+
+    this.dataService.triggerEvaluation(data).subscribe(
+      (res) => {
+        this.handleResponse(res);
+      },
+      (err) => {
+        this.isLoading = false;
+        console.log(err);
+      }
+    );
+  }
+
+  private triggerCustomSequence(): void {
+    const data: TriggerSequenceData = {
+      project: this.projectName || '',
+      stage: this.selectedStage || '',
+      service: this.selectedService || '',
+    };
+
+    if (this.customFormData.labels && this.customFormData.labels.trim() !== '') {
+      data.labels = this.parseLabels(this.customFormData.labels);
+    }
+
+    this.dataService.triggerCustomSequence(data, this.customFormData.sequence || '').subscribe(
+      (res) => {
+        this.handleResponse(res);
+      },
+      (err) => {
+        this.isLoading = false;
+        console.log(err);
+      }
+    );
+  }
+
+  private handleResponse(response: TriggerResponse): void {
+    AppUtils.createTimer(1000, 200)
+      .pipe(takeUntil(this.unsubscribe$))
+      .subscribe(() => {
+        this.dataService.getEventByContext(response.keptnContext).subscribe(
+          (events) => {
+            if (events.length > 0) {
+              this.isLoading = false;
+              this.unsubscribe$.next();
+              this.router.navigate([
+                '/project',
+                this.projectName,
+                'sequence',
+                response.keptnContext,
+                'stage',
+                this.selectedStage,
+              ]);
+            }
+          },
+          () => {
+            // Gracefully fail - and just navigate to sequences
+            this.isLoading = false;
+            this.unsubscribe$.next();
+            this.router.navigate(['/project', this.projectName]);
+          }
+        );
+      });
```

Comment:
As discussed, we need to check the GET `/sequence` endpoint instead of the GET `/event` endpoint, because the data-flow there is obviously async.
Either use `dataService.loadSequences()` or change `apiService.getTraces()` to `apiService.getSequences()` in `dataService.getEventByContext()`